torchtune/utils/metric_logging.py

```diff
@@ -160,27 +174,63 @@ def __init__(
                 "Alternatively, use the ``StdoutLogger``, which can be specified by setting metric_logger_type='stdout'."
             ) from e
         self._wandb = wandb
-        self._wandb.init(
-            project=project,
-            entity=entity,
-            group=group,
-            reinit=True,
-            resume="allow",
-            config=kwargs,
-        )
+
+        # logging strategy options are "main", "node", "all"
+        self.log_strategy = log_strategy
+        self.world_size, self.rank = get_world_size_and_rank()
+        self.local_rank = _get_local_rank()
+        self.local_rank = 0 if self.local_rank is None else self.local_rank
+        
+        if (
+                (self.log_strategy == "main" and self.rank == 0) 
+                or (self.log_strategy == "node" and self.local_rank == 0)
+                or self.log_strategy == "all"
```

Comment:
I'm inclined to make this a quick private method (e.g., `_should_log()`) since the logic here is more complex, that way you can test this logic in an isolated way